browser/scripts/ui-core-statestore.js

```diff
@@ -0,0 +1,188 @@
+var uiMode = {
+	build: 	'build',
+	program: 'program'
+}
+var uiModifyMode = {
+	move: 	'translate',
+	rotate: 'rotate',
+	scale: 	'scale'
+}
+
+/**
+ * first-level properties e.g. .mode, .visibility emit events upon changing
+ * second-level properties through .visibility, .panelStates and .context also emit events
+ * if persistentStorageRef, except for a number of ignored properties, the object will persist
+ * @param persistentStorageRef (store or recall state)
+ */
+var UiStateStore = function(persistentStorageRef, context) {
+	EventEmitter.apply(this, arguments)
+	var that = this
+
+	var a = function(o, props, callback) { // ES5
+		// top-level setter (e.g. obj.property = value) generates a changed:property event
+		o._ = props
```

Comment:
it could. may I leave it as it is? my reasons are 1] outside of reading the actual code I specifically want to discourage its use, and 2] I find a `state._.key` easier to read than `state._properties.key` (after all _properties is a property).